imperial_coldfront_plugin/views.py

```diff
@@ -51,3 +61,93 @@ def group_members_view(request: HttpRequest, user_pk: int) -> HttpResponse:
     group_members = GroupMembership.objects.filter(owner=user)
 
     return render(request, "group_members.html", {"group_members": group_members})
+
+
+@require_POST
+@login_required
+def invite_to_group(request: HttpRequest) -> HttpResponse:
+    """Invite an individual to a group.
+
+    Args:
+        request: The HTTP request object containing metadata about the request.
+    """
+    signer = TimestampSigner()
+    invitee_email = request.POST.get("invitee_email")
+
+    # Sign invitation.
+    token = signer.sign_object(
+        {
+            "inviter_pk": request.user.pk,
+            "invitee_email": invitee_email,
+        }
+    )
+
+    invite_url = request.build_absolute_uri(
+        reverse("imperial_coldfront_plugin:accept_invite", args=[token])
+    )
+
+    # Send invitation via email.
+    send_mail(
+        "You've been invited to a group",
+        f"Click the following link to accept the invitation: {invite_url}",
+        settings.DEFAULT_FROM_EMAIL,
+        [invitee_email],
+    )
+
+    return render(
+        request=request,
+        context={
+            "invitee_email": invitee_email,
+            "token": token,
+            "invite_url": invite_url,
+        },
+        template_name="imperial_coldfront_plugin/invite_to_group.html",
+    )
+
+
+@login_required
+def accept_invite(request: HttpRequest, token: str) -> HttpResponse:
+    """Accept invitation to a group.
+
+    Args:
+        request: The HTTP request object containing metadata about the request.
+        token: The token that was sent to the invitee.
+    """
+    signer = TimestampSigner()
+
+    # Validate token.
+    try:
+        invite = signer.unsign_object(token, max_age=86400)
+    except SignatureExpired:
+        return HttpResponseBadRequest("Expired token")
+    except BadSignature:
+        return HttpResponseBadRequest("Bad token")
+
+    # Check the correct user is using the token.
+    if invite["invitee_email"] != request.user.email:
+        return HttpResponseForbidden(
+            "The invite token is not associated with this email address."
+        )
+
+    # Update group membership in the database.
+    group = ResearchGroup.objects.get(owner__pk=invite["inviter_pk"])
+    GroupMembership.objects.create(group=group, member=request.user)
+
+    return render(
+        request=request,
+        context={
+            "inviter_pk": invite["inviter_pk"],
+            "invitee_email": invite["invitee_email"],
+        },
+        template_name="imperial_coldfront_plugin/accept_invite.html",
+    )
+
+
+@login_required
+def index(request: HttpRequest) -> HttpResponse:
+    """Render the index page.
+
+    Args:
+        request: The HTTP request object containing metadata about the request.
+    """
+    return render(request=request, template_name="imperial_coldfront_plugin/index.html")
```

Comment:
The usual Django pattern is to handle the display and processing of a form in a single view function. The example linked in the Django docs shows how it works.